### PR TITLE
Replace rounding_mode="true" with rounding_mode=None

### PR DESCRIPTION
**Pull request by peterbell10**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #55225 grad_fn getter for optional strings
* #50438 Support JIT shape analysis for div with rounding_mode argument
* #51989 Sparse support for division rounding_mode argument
* #51792 Vectorize floating point floor_divide
* **#51988 Replace rounding_mode="true" with rounding_mode=None**

Differential Revision: [D27561817](https://our.internmc.facebook.com/intern/diff/D27561817)
### `torch.divide` with `rounding_mode='true'` is replaced with `rounding_mode=None`

The `torch.divide`'s undocumented `rounding_mode='true'` option has been removed, and instead `rounding_mode=None` can be passed to indicate no rounding should take place. This is equivalent to omitting the argument entirely.

Example:

Version 1.8.1:
```
>>> a, b = torch.full((2,), 4.2), torch.full((2,), 2)
>>> torch.divide(a, b, rounding_mode='true')
tensor([2.1000, 2.1000])
```

Version 1.9.0:
```
>>> a, b = torch.full((2,), 4.2), torch.full((2,), 2)
>>> torch.divide(a, b, rounding_mode=None)
tensor([2.1000, 2.1000])
```

### CODE REVIEW

**Review by mruberry — APPROVED** [review]
Awesome! Thanks for the quick fix, @peterbell10!

**Comment by peterbell10** [comment]
@mruberry PTAL. Rebased, test are passing, and sent PR updating XLA pytorch/xla#2859.

File: test/test_autograd.py
Comment (mruberry) [inline test/test_autograd.py:4536]: Why change this to be a linalg.qr test?

**Review by mruberry — APPROVED** [review]
I just have one small question. This is awesome, @peterbell10! Thank you for picking up the initiatives on following-through with these deprecations.

@JackCaoG let me know if/when you're ready to merge the XLA side of this, then we can coordinate a merge like we typically do.

**Comment by mruberry** [comment]
Imported to hit internal tests. There are already some uses of rounding_mode within FB! But I don't think any were using the undocumented "true" mode.

File: test/test_autograd.py
Comment (peterbell10) [reply test/test_autograd.py:4536]: Because it doesn't have a `str` argument any more. See #55225 where I add it back.

**Comment by JackCaoG** [comment]
@mruberry Peter submitted https://github.com/pytorch/xla/pull/2859 and it is ready now. I will merge xla pr once tis one is merged.

**Comment by mruberry** [comment]
@JackCaoG Awesome! Beginning the land process now. I'll update here when it finishes.

**Pull request closed without merging**
